util/timeutil/time.go

```diff
@@ -0,0 +1,156 @@
+}
+
+// GetSystemTZ reads system timezone from `TZ`, the path of the soft link of `/etc/localtime`. If both of them are failed, system timezone will be set to `UTC`.
+// It is exported because we need to use it during bootstap stage. And it should be only used at that stage.
+func GetSystemTZ() string {
+	// consult $TZ to find the time zone to use.
+	// no $TZ means use the system default /etc/localtime.
+	// $TZ="" means use UTC.
+	// $TZ="foo" means use /usr/share/zoneinfo/foo.
+	tz, ok := syscall.Getenv("TZ")
+	switch {
+	case !ok:
+		path, err1 := filepath.EvalSymlinks("/etc/localtime")
+		if err1 == nil {
+			name, err2 := getTZNameFromFileName(path)
+			if err2 == nil {
+				return name
+			}
+			log.Errorln(err2)
+		}
+		log.Errorln(err1)
+	case tz != "" && tz != "UTC":
+		for _, source := range zoneSources {
+			if _, err := os.Stat(source + tz); err == nil {
+				return tz
+			}
+		}
+	}
+	return "UTC"
+}
+
+// getTZNameFromFileName gets IANA timezone name from zoneinfo path.
+// TODO It will be refined later. This is just a quick fix.
+func getTZNameFromFileName(path string) (string, error) {
+	// phase1 only support read /etc/localtime which is a softlink to zoneinfo file
+	substr := "zoneinfo"
+	if strings.Contains(path, substr) {
+		idx := strings.Index(path, substr)
+		return string(path[idx+len(substr)+1:]), nil
+	}
+	return "", errors.New(fmt.Sprintf("path %s is not supported", path))
+}
+
+// Local returns time.Local's IANA timezone name.
+func Local() *time.Location {
+	loc, err := LoadLocation(LocalStr)
+	if err != nil {
+		return time.Local
+	}
+	return loc
+}
+
+// getLoc first trying to load location from a cache map. If nothing found in such map, then call
+// `time.LocadLocation` to get a timezone location. After trying both way, an error wil be returned
+//  if valid Location is not found.
+func (lm *locCache) getLoc(name string) (*time.Location, error) {
+	if name == "System" {
+		return time.Local, nil
+	}
+	lm.RLock()
+	if v, ok := lm.locMap[name]; ok {
+		lm.RUnlock()
+		return v, nil
+	}
+
+	if loc, err := time.LoadLocation(name); err == nil {
+		lm.RUnlock()
+		lm.Lock()
+		lm.locMap[name] = loc
+		lm.Unlock()
+		return loc, nil
+	}
+
+	lm.RUnlock()
+	return nil, fmt.Errorf("invalid name for timezone %s", name)
+}
+
+// LoadLocation loads time.Location by IANA timezone time.
+func LoadLocation(name string) (*time.Location, error) {
+	return LocCache.getLoc(name)
+}
+
+// Zone returns the current timezone name and timezone offset in seconds.
+// In compatible with MySQL, we change `Local` to `System`.
+// TODO: Golang team plan to return system timezone name intead of
+// returning `Local` when `loc` is `time.Local`. We need keep an eye on this.
+func Zone(loc *time.Location) (string, int64) {
+	_, offset := time.Now().In(loc).Zone()
+	var name string
+	name = loc.String()
+	// when we found name is "Local", we have no chice but push down
+	// "System" to tikv side.
+	if name == "Local" {
```

Comment:
After this change, the name should never be `Local` or `System`.

Reply:
It is possible when we are on testing stage.